### ISSUE TITLE
Add cohen's kappa metric score

### ISSUE THREAD

**Issue by innat** [keras-team/keras-cv#443]
**Short Description**

This function computes Cohen’s kappa, a score that expresses the level of agreement between two annotators on a classification problem. It is defined as

![image](https://user-images.githubusercontent.com/17668390/169282603-96865edb-8ac9-430b-a31c-128e6b0ff34c.png)

where `p_0`  is the empirical probability of agreement on the label assigned to any sample (the observed agreement ratio), and `p_e`  is the expected agreement when both annotators assign labels randomly `p_e`.  is estimated using a per-annotator empirical prior over the class labels

[src.](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.cohen_kappa_score.html)

**Papers**

- https://journals.sagepub.com/doi/10.1177/001316446002000104
- https://en.wikipedia.org/wiki/Cohen%27s_kappa

**Existing Implementations**
- https://scikit-learn.org/stable/modules/generated/sklearn.metrics.cohen_kappa_score.html
- https://github.com/tensorflow/addons/blob/master/tensorflow_addons/metrics/cohens_kappa.py

**Comment by bhack** [comment]
Is this a CV only metric?

**Comment by innat** [comment]
ok, not sure, it may be general.
I mostly used it in ordinal regression-related CV tasks.
Curious, if it's general, what should be the approach to include this?

**Comment by bhack** [comment]
>  Curious, if it's general, what should be the approach to include this?

I don't know https://github.com/keras-team/keras-cv/pull/30#issuecomment-1008685239

**Comment by innat** [comment]
hmm, that's an issue. 
@LukeWood thoughts?

**Comment by bhack** [comment]
@innat We had a similar discussion at https://github.com/keras-team/keras-cv/issues/137#issuecomment-1042903162

**Comment by bhack** [comment]
See also our threads at:
https://github.com/keras-team/keras/issues/16181#issuecomment-1064366843
https://github.com/keras-team/keras-cv/issues/137#issuecomment-1043585248

**Comment by innat** [comment]
It's really an intricate issue and needs separate discussion. @LukeWood could you please give some feedback on this? 

cc. @fchollet @qlzh727

**Comment by innat** [comment]
cc. @LukeWood 
A gentle reminder here.
